File: worker/worker-impl/src/main/java/com/masterello/worker/listener/CategoriesChangedEventListener.java
package com.masterello.worker.listener;

import com.masterello.category.event.CategoriesChangedEvent;
import com.masterello.worker.validator.ServiceValidator;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.jetbrains.annotations.NotNull;
import org.springframework.context.ApplicationListener;
import org.springframework.stereotype.Component;

@Component
@Slf4j
@RequiredArgsConstructor
public class CategoriesChangedEventListener implements ApplicationListener<CategoriesChangedEvent> {


    private final ServiceValidator serviceValidator;

    @Override
    public void onApplicationEvent(@NotNull CategoriesChangedEvent event) {
        serviceValidator.init();
    }
}

File: category/category-impl/src/main/kotlin/com/masterello/category/listener/CategoriesChangedEventListener.kt
package com.masterello.category.listener

import com.masterello.category.event.CategoriesChangedEvent
import com.masterello.commons.async.MasterelloEventListener
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.cache.annotation.CacheEvict
import org.springframework.stereotype.Component

@Component
class CategoriesChangedEventListener : MasterelloEventListener<CategoriesChangedEvent>() {

    private val log = KotlinLogging.logger {}

    @CacheEvict("getAllCategories", allEntries = true)
    override fun onApplicationEvent(event: CategoriesChangedEvent) {
        super.onApplicationEvent(event)
    }

    override fun processEvent(event: CategoriesChangedEvent?) {
        log.info { "Evict getAllCategories cache" }
    }
}